core/src/main/java/org/elasticsearch/index/query/ConstantScoreQueryBuilder.java

```diff
@@ -66,20 +70,78 @@ public ConstantScoreQueryBuilder boost(float boost) {
         return this;
     }
 
+    /**
+     * @return the boost factor
+     */
+    public float boost() {
+        return this.boost;
+    }
+
     @Override
     protected void doXContent(XContentBuilder builder, Params params) throws IOException {
         builder.startObject(NAME);
         builder.field("filter");
-        filterBuilder.toXContent(builder, params);
+        queryBuilder.toXContent(builder, params);
+        builder.field("boost", boost);
+        builder.endObject();
+    }
+
+    @Override
+    public Query toQuery(QueryParseContext parseContext) throws QueryParsingException, IOException {
+        Query innerFilter = queryBuilder.toQuery(parseContext);
+        if (innerFilter == null ) {
+            // no inner query, we ignore this and return no docs
+            return Queries.newMatchNoDocsQuery();
```

Comment:
ok I do not like the way we use null here but I see the complication of changing this now. Let's leave it as is then and move on ;)